src/containers/my-quizzes/create-or-edit-quiz-container/CreateOrEditQuizContainer.tsx

```diff
@@ -82,90 +78,84 @@ const CreateOrEditQuizContainer = ({
   }
 
   const handleResponse = () => {
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.success,
-        message: id
-          ? 'myResourcesPage.quizzes.successEditedQuiz'
-          : 'myResourcesPage.quizzes.successAddedQuiz'
-      })
-    )
+    handleAlert({
+      severity: snackbarVariants.success,
+      message: id
+        ? 'myResourcesPage.quizzes.successEditedQuiz'
+        : 'myResourcesPage.quizzes.successAddedQuiz'
+    })
     navigateToQuizzesTab()
   }
 
-  const onResponseError = (error?: ErrorResponse) => {
-    const errorKey = getErrorKey(error)
-
-    dispatch(
-      openAlert({
-        severity: snackbarVariants.error,
-        message: error
-          ? {
-              text: errorKey,
-              options: {
-                message: getErrorMessage(error.message)
-              }
-            }
-          : errorKey
-      })
-    )
-  }
-
   const createQuizService = useCallback(
-    (data?: CreateQuizParams) => ResourceService.addQuiz(data),
+    (data: CreateQuizParams) => ResourceService.addQuiz(data),
     []
   )
 
-  const { fetchData: addNewQuiz } = useAxios<Quiz, CreateQuizParams>({
-    service: createQuizService,
-    fetchOnMount: false,
-    defaultResponse,
-    onResponse: handleResponse,
-    onResponseError
+  const { mutate: fetchAddQuiz } = useMutation({
+    mutationFn: createQuizService,
+    onSuccess: handleResponse,
+    onError: handleErrorAlert
   })
 
   const editQuizService = useCallback(
-    (params?: UpdateQuizParams) => ResourceService.editQuiz(params),
-    []
+    async (data: UpdateQuizParams) => {
+      if (id) {
+        await ResourceService.editQuiz(data)
+      }
+    },
+    [id]
   )
 
-  const { fetchData: fetchEditedQuiz } = useAxios<null, UpdateQuizParams>({
-    service: editQuizService,
-    fetchOnMount: false,
-    defaultResponse: null,
-    onResponse: handleResponse,
-    onResponseError
+  const { mutate: fetchEditedQuiz } = useMutation({
+    queryKey: ['quiz', id],
+    mutationFn: editQuizService,
+    onSuccess: handleResponse,
+    onError: handleErrorAlert
   })
 
-  const getQuiz = (id?: string): Promise<AxiosResponse> => {
-    return ResourceService.getQuiz(id)
-  }
+  const getQuiz = useCallback(() => {
+    if (id) {
+      return ResourceService.getQuizQuery(id)
+    }
+    return defaultResponse
```

Comment:
Let's consider that we should have an `id` here (default value as in previous comment)
```suggestion
      return ResourceService.getQuizQuery(id)
```